### PR TITLE
KEEP-130: Venue Notification

### PR DESCRIPTION
**Pull request by jfabian06ph**
#### Description

[KEEP-130](https://ordermentum.atlassian.net/browse/KEEP-130)

This changes will enable to:
1. Fetch OMNotifications
2. Mark OMNotification as read

----
#### Changes
• Add VenueNotificationsResponse
• Add NotificationsClient class

----

### CODE REVIEW

File: OrdermentumSDK/Models/Profile.swift [hunk 1]
```diff
@@ -34,6 +34,11 @@ public struct UserProfile: Codable {
     public var verifiedAt: String = ""
 }
 
+public struct VenueNotificationResponse {
```
Comment (bstillitano) [inline OrdermentumSDK/Models/Profile.swift:37]: Change to `VenueNotificationsResponse`

File: OrdermentumSDK/Models/Profile.swift [hunk 2]
```diff
@@ -220,6 +237,21 @@ extension SupplierNotifications {
     }
 }
 
+extension VenueNotifications {
```
Comment (bstillitano) [inline OrdermentumSDK/Models/Profile.swift:240]: Declare the extension conformance type here as `Decodable`. Remove `Codable` from the `struct` declaration if we're not encoding the object anywhere.

File: OrdermentumSDK/Routers/NotificationsRouter.swift [hunk 3]
```diff
@@ -13,6 +13,8 @@ public enum NotificationsRouter: URLRequestConvertible {
     //Routes
     case updateNotificationSetting(String, String, [String: Bool])
     case updateAllNotificationSettings(String, SupplierNotifications)
+    case getNotification(String)
```
Comment (bstillitano) [inline OrdermentumSDK/Routers/NotificationsRouter.swift:16]: If this returns an array of notifications, please name it `getNotifications`

File: OrdermentumSDK/Routers/NotificationsRouter.swift [hunk 4]
```diff
@@ -13,6 +13,8 @@ public enum NotificationsRouter: URLRequestConvertible {
...
+    case readNotification(String)
```
Comment (bstillitano) [inline OrdermentumSDK/Routers/NotificationsRouter.swift:17]: Give this a more meaningful name such as `markNotificationAsRead`

File: OrdermentumSDK/Models/Profile.swift [hunk 5]
```diff
@@ -70,6 +75,18 @@ public struct Notifications: Codable {
     public var id: SupplierNotifications = SupplierNotifications()
 }
 
+public struct VenueNotifications {
```
Comment (bstillitano) [inline OrdermentumSDK/Models/Profile.swift:78]: Please change this to `VenueNotification`

File: OrdermentumSDK/NotificationsClient.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            return try URLEncoding.default.encode(request, with: parameters)
+        default:
+            return try JSONEncoding.default.encode(request, with: body?.toParameters())
+        }
+    }
+}
+
+public enum NotificationsClientURL {
+    public static let rootNotificationURL = "https://notifications.ordermentum-sandbox.com/v2/"
+}
```
Comment (bstillitano) [inline OrdermentumSDK/NotificationsClient.swift:61]: Only sandbox URL is supplier here? Also it should be clear that it is sandbox and not prod with an appropriate variable name.

File: OrdermentumSDK/Routers/NotificationsRouter.swift [hunk 7]
```diff
@@ -49,13 +61,20 @@ public enum NotificationsRouter: URLRequestConvertible {
...
     
     //Builder
     public func asURLRequest() throws -> URLRequest {
-        return try Client.instance.urlRequest(path: path, method: method, parameters: parameters, body: body)
+        switch self {
+        case .getNotifications, .markNotificationAsRead:
+            return try NotificationsClient.instance.urlRequest(path: path, method: method, parameters: parameters, body: body)
+        default:
+            return try Client.instance.urlRequest(path: path, method: method, parameters: parameters, body: body)
+        }
```
Comment (bstillitano) [inline OrdermentumSDK/Routers/NotificationsRouter.swift:78]: Why are we switching here? If different calls use a different client we should have different services for each.

File: OrdermentumSDK/Services/NotificationsService.swift [hunk 8]
```diff
@@ -39,4 +39,25 @@ public class NotificationsService {
...
+            completion(result, responseObject, errorObject)
+        }
+    }
+
+    public func readNotification(notificationId: String, completion: @escaping (Bool, ErrorResponse?) -> ()) {
+        //Build Route
+        let route = NotificationsRouter.markNotificationAsRead(notificationId) as URLRequestConvertible
+
+        //Call API
+        Service<VenueNotificationsResponse, ErrorResponse>().request(route: route) { (result, responseObject, errorObject) in
```
Comment (bstillitano) [inline OrdermentumSDK/Services/NotificationsService.swift:58]: Should this be returning an `EmptyDecodable` instead?

File: OrdermentumSDK/NotificationsClient.swift (hunk 6 above)
Comment (jfabian06ph) [reply OrdermentumSDK/NotificationsClient.swift:62]: Added NotificationsClient and finalized NotificationsClientURL

**Comment by bstillitano** [comment]
Test failing but not a blocker for this PR

**Pull request merged**